server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -250,6 +252,45 @@ void dequeueShouldWorkAfterNetworkOutage() throws Exception {
                 .containsExactly(name1, name2, name3);
         }
 
+        @Test
+        void messagesShouldSurviveRabbitMQRestart() throws Exception {
+            String name1 = "myMail1";
+            String name2 = "myMail2";
+            String name3 = "myMail3";
+            Flux<MailQueue.MailQueueItem> dequeueFlux = Flux.from(getMailQueue().deQueue());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .build());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name2)
+                .build());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name3)
+                .build());
+
+            Thread.sleep(2000);
```

Comment:
I'm ok waiting but can you add a comment explaining what we are waiting for?

Also, I've been seeing some rabbit restart without a "wait" before them. Maybe that's some potential bugs?

Reply:
I though that the enqueue was  an async process. But it is synchronous in fact. I removed the sleep.